src/block_neural.rs

```diff
@@ -282,9 +241,25 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockNeuronLayer<L> {
         debug_assert!(self.output_offset != usize::MAX);
         debug_assert!(self.input_offset != usize::MAX);
 
+        // If we are in pure prediction mode (
+        let dropout_inv = match update {
+            true => self.dropout_inv,
+            false => 1.0,
+        };
+
+        self.internal_forward(pb, dropout_inv);
+
+        block_helpers::forward_backward(further_blocks, fb, pb, update);
+
         unsafe {
-            let bias_offset = self.num_inputs * self.num_neurons;
-            {
+            if update && self.neuron_type == NeuronType::WeightedSum {
+                // first we need to initialize inputs to zero
+                // TODO - what to think about this buffer
```

Comment:
Not sure what this means

Reply:
That's a leftover from before. I will remove it as the comment doesn't add any value.